apollo-http-cache/src/main/kotlin/com/apollographql/apollo3/cache/http/DiskLruHttpCache.kt

```diff
@@ -104,6 +137,51 @@ class DiskLruHttpCache(private val fileSystem: FileSystem, private val directory
     }
   }
 
+  /**
+   * A [BufferedSource] that writes to the given cache sink as it is read.
+   */
+  private class WriteToCacheSource(
+      private val originalBody: Source,
+      private val cacheSink: Sink,
```

Comment:
Nit: what about proxy? 

```suggestion
  private class ProxySource(
      private val originalSource: Source,
      private val sink: Sink,
```